flang/lib/Optimizer/CodeGen/TypeConverter.h

```diff
@@ -155,6 +158,27 @@ class LLVMTypeConverter : public mlir::LLVMTypeConverter {
   // i64 can be used to index into aggregates like arrays
   mlir::Type indexType() { return mlir::IntegerType::get(&getContext(), 64); }
 
+  // fir.type<name(p : TY'...){f : TY...}>  -->  llvm<"%name = { ty... }">
+  llvm::Optional<LogicalResult>
+  convertRecordType(fir::RecordType derived,
+                    SmallVectorImpl<mlir::Type> &results,
+                    ArrayRef<mlir::Type> callStack) {
+    auto name = derived.getName();
+    auto st = mlir::LLVM::LLVMStructType::getIdentified(&getContext(), name);
+    if (llvm::count(callStack, derived) > 1) {
```

Comment:
Algo comment: it's too bad this is now a hard `O(n)` instead of a short-circuiting test like `any` or `find`. On the other hand, these stacks should, typically, have small values of `n`.

Reply:
I am assuming since the stack has small values of `n` this should be OK. Code is also straightforward to read this way. Would it be OK to retain llvm::count?

Reply:
I think it is ok. But I think it deserves a comment-note, since it "looks wrong" (at least to me). Something like we decided to use an O(n) here because n is (or should be) very small.

Reply:
Added comment as suggested.
I will try to convert this to a short-circuiting search upstream for all such uses of count in type conversion (in flang and mlir) and then merge it down in a separate PR